### PR TITLE
Pile of commits that should be reviewed and merged

### PR DESCRIPTION
**Pull request by thiblahute**
And misc enhancements

### CODE REVIEW

File: hotdoc/core/extension.py [hunk 1]
```diff
@@ -441,9 +450,16 @@ def _resolve_placeholder(self, tree, name, include_paths):
...
-        elif self.smart_index and override_path in self._get_all_sources():
-            path = find_file('%s.markdown' % name, include_paths)
-            return path or True, None
+
+        if self.smart_index:
+            for path in OrderedSet([self.__package_root]) | self.source_roots:
+                possible_path = os.path.join(path, name)
+                if possible_path in self._get_all_sources():
+                    override_path = find_file('%s.markdown' % name, include_paths)
+                    debug(possible_path)
```
Comment (MathieuDuponchelle) [inline hotdoc/core/extension.py]: Can you add more context to that debug please?

File: hotdoc/core/extension.py [hunk 2]
```diff
@@ -482,7 +498,7 @@ def __find_package_root(self):
         if self.__package_root:
             return
 
-        commonprefix = os.path.commonprefix(list(self._get_all_sources()))
+        commonprefix = os.path.commonprefix(list(self._get_all_sources()) + list(self.source_roots))
```
Comment (MathieuDuponchelle) [inline hotdoc/core/extension.py]: Do we still need `__package_root` at all if source_roots have been explicitly specified?

File: hotdoc/core/comment.py [hunk 3]
```diff
@@ -84,20 +83,36 @@ def __init__(self, name=u'', title=None, params=None, filename=u'',
         self.initial_col_offset = 0
         self.annotations = annotations or {}
 
-        meta = meta or {}
+        meta = self.__cleanup_meta(meta)
```
Comment (MathieuDuponchelle) [inline hotdoc/core/comment.py]: Commit message is unrelated, please split that up with a proper message :)

File: hotdoc/extensions/gi/gi_extension.py [hunk 4]
```diff
@@ -233,7 +234,10 @@ def _get_smart_key(self, symbol):
                                 super()._get_smart_key(symbol))
 
     def _get_all_sources(self):
-        return [s for s in self.c_sources if s.endswith('.h')]
+        if not self.__all_sources:
+            self.__all_sources = list({s.replace('.c', '.h') for s in self.c_sources})
```
Comment (MathieuDuponchelle) [inline hotdoc/extensions/gi/gi_extension.py]: That's very weird, did we end up finding another way to do that? If not, we probably should discuss that :)

File: hotdoc/core/extension.py (hunk 2 above)
Comment (thiblahute) [reply hotdoc/core/extension.py]: Yes, everything is still relative to it internally, user doesn't have to know.

File: hotdoc/extensions/gi/gi_extension.py (hunk 4 above)
Comment (thiblahute) [reply hotdoc/extensions/gi/gi_extension.py]: Fixed the way we agreed on (c files keep being referenced as `.c`).

**Comment by thiblahute** [comment]
Also splited the porter commit with a proper commit to "cleanup metas"
